dev/devicelab/bin/tasks/ios_app_with_extensions_test.dart

```diff
@@ -33,22 +35,53 @@ Future<void> main() async {
       await inDirectory(projectDir, () async {
         await flutter(
           'build',
-          options: <String>['ios', '--no-codesign'],
+          options: <String>['ios', '--no-codesign', '--release'],
         );
       });
 
-      final bool appReleaseBuilt = exists(Directory(path.join(
+      final String appBundle = Directory(path.join(
         projectDir.path,
         'build',
         'ios',
         'iphoneos',
         'Runner.app',
-      )));
+      )).path;
 
-      if (!appReleaseBuilt) {
-        return TaskResult.failure(
-            'Failed to build flutter iOS app with WatchOS companion in release mode.');
-      }
+      final String appFrameworkPath = path.join(
+        appBundle,
+        'Frameworks',
+        'App.framework',
+        'App',
+      );
+      final String flutterFrameworkPath = path.join(
+        appBundle,
+        'Frameworks',
+        'Flutter.framework',
+        'Flutter',
+      );
+
+      checkDirectoryExists(appBundle);
+      await _checkFlutterFrameworkArchs(appFrameworkPath, isSimulator: false);
+      await _checkFlutterFrameworkArchs(flutterFrameworkPath, isSimulator: false);
+
+      // Check the watch extension framework is in place with the expected watch archs.
+      final String watchExtensionFrameworkPath = path.join(
+        appBundle,
+        'Watch',
+        'watch.app',
+        'PlugIns',
+        'watch Extension.appex',
+        'Frameworks',
+        'EFQRCode.framework',
+        'EFQRCode',
+      );
+      _checkWatchExtensionFrameworkArchs(watchExtensionFrameworkPath);
```

Comment:
This is the new check that these plugin frameworks haven't been `lipo`d to the iOS ARCHS.